.github/actions/check-dependent-rust-repo/action.yml

```
@@ -0,0 +1,139 @@
+      required: true
+    githubToken:
+      description: A GitHub token for API access.
+      required: true
+    submodulePath:
+      description: The path to the submodule we are going to try upreving
+      required: true
+    buildCmd:
+      description: The command to build the remote repo
+      required: true
+
+runs:
+  using: composite
+  steps:
+  - name: Setup Rust
+    uses: ./mobilecoin/.github/actions/setup-rust
+
+  # Checkout the repository we are going to try upreving in
+  - uses: actions/checkout@v3
+    with:
+      repository: ${{ inputs.remoteRepo }}
+      submodules: 'recursive'
+      path: workdir-${{ inputs.id }}
+
+  # We use the cache to store state about previous runs of this action
+  - name: Cache state
+    id: cache-state
+    uses: actions/cache@v3
+    with:
+      path: state-${{ inputs.id }}
+      key: dep-v1-${{ runner.os }}-${{ github.ref }}-${{ inputs.id }}-${{ github.sha }}
+      restore-keys: |
+        dep-v1-${{ runner.os }}-${{ github.ref }}-${{ inputs.id }}-
+
+  - name: Build outside repo with current branch
+    shell: bash
+    run: |
+      set -x
+      cd workdir-${{ inputs.id }}
+
+      STATE_DIR=$GITHUB_WORKSPACE/state-${{ inputs.id }}
+      mkdir -p $STATE_DIR
+      CUR_BRANCH=${GITHUB_REF//\//_}
+
+      # If this file exists, it means we have failed building this branch at least once.
+      STATE_FILE="$STATE_DIR/$CUR_BRANCH-failed"
+
+      # Attempt to uprev the submodule to the latest revision in the current PR this action is running in
+      cd ${{ inputs.submodulePath }}
+      #git remote set-url origin ${{ github.repositoryUrl }} - this is git:// and GHA doesn't seem to like it
+      git remote set-url origin https://github.com/${{ github.repository }}
+      git fetch origin ${{ github.ref }}:uprev-test
+      git checkout uprev-test
+      git log -1 --format='%H'
+      cd ..
+
+      # Try to build
+      cat <<'EOF' > build-cmd
+      ${{ inputs.buildCmd }}
+      EOF
+
+      bash build-cmd && {
+        # Build succeeded, lets see if this is a new development.
+        if [ -f "$STATE_FILE" ]; then
+          echo "We previously failed building, but this run succeeded :)"
+          rm $STATE_FILE
+
+          # This lets the steps below know that we fixed a previously failed build.
+          echo "UPREV_FIXED=1" >> "$GITHUB_ENV"
+        fi
+       } || {
+        # We failed building, so keep track of that in the env. This lets the steps below
+        # take action based on whether we failed or not.
+        echo "UPREV_FAILED=1" >> "$GITHUB_ENV"
+
+        # If this a new failure?
+        if [ -f "$STATE_FILE" ]; then
+          echo "Not the first time this branch failed, so we won't be posting an update."
+          echo "NEW_FAILURE=0" >> "$GITHUB_ENV"
+        else
+          echo "Detected a new failure, indicating by writing to $STATE_FILE"
+          echo > $STATE_FILE
+          echo "NEW_FAILURE=1" >> "$GITHUB_ENV"
+        fi
+      }
```

Consider using `if` here
```suggestion
      if bash build-cmd; then
        # Build succeeded, lets see if this is a new development.
        if [ -f "$STATE_FILE" ]; then
          echo "We previously failed building, but this run succeeded :)"
          rm $STATE_FILE

          # This lets the steps below know that we fixed a previously failed build.
          echo "UPREV_FIXED=1" >> "$GITHUB_ENV"
        fi
       else
        # We failed building, so keep track of that in the env. This lets the steps below
        # take action based on whether we failed or not.
        echo "UPREV_FAILED=1" >> "$GITHUB_ENV"

        # If this a new failure?
        if [ -f "$STATE_FILE" ]; then
          echo "Not the first time this branch failed, so we won't be posting an update."
          echo "NEW_FAILURE=0" >> "$GITHUB_ENV"
        else
          echo "Detected a new failure, indicating by writing to $STATE_FILE"
          echo > $STATE_FILE
          echo "NEW_FAILURE=1" >> "$GITHUB_ENV"
        fi
      fi
```